tools/pot/openvino/tools/pot/engines/ac_engine.py

```diff
@@ -123,14 +125,31 @@ def predict(self, stats_layout=None, sampler=None, stat_aliases=None,
         callback_layout, stat_names_aliases = {}, {}
         # add outputs for activation statistics collection
         if stats_layout is not None:
-            model_with_stat_op, nodes_name, output_to_node_names = self._statistic_graph_builder.\
+            model_with_stat_op, nodes_names_map, output_to_node_names = self._statistic_graph_builder.\
                 insert_statistic(copy.deepcopy(self._nx_model),
                                  stats_layout, stat_aliases)
             self.set_model(model_with_stat_op)
-            add_outputs(self._model, nodes_name)
+            
+            for model in self._model:
+                cast_friendly_names(model['model'].outputs)
+
+            outputs_per_model = add_outputs(self._model, nodes_names_map)
+            for model_name, outputs_data in outputs_per_model.items():
+                add_tensor_names(outputs_data, nodes_names_map[model_name].keys())
+
             self._model_evaluator.load_network(self._model)
 
-            model_output_names = [out for m_dict in self._model for out in m_dict['model'].outputs.keys()]
+            model_output_names = []
+            model_outputs = []
+            for model in self._model:
+                model_outputs.extend(model['model'].outputs)
+            for ng_output in model_outputs:
+                model_output_names.extend(list(ng_output.get_tensor().get_names()))
+
+            nodes_name = []
+            for names_map in nodes_names_map.values():
```

Comment:
I noticed some of the code that is responsible for tensor names, etc. is duplicated in both Engines (AC and IE). Can we extract it in a meaningful method and reuse?

Reply:
Yes, working on it. Thanks for the notice.